public/components/integrations/components/setup_integration.tsx

```diff
@@ -132,6 +146,21 @@ const prepareQuery = (query: string, config: IntegrationSetupInputs): string =>
   return queryStr;
 };
 
+/**
+ * Handles the integration setup process based on the connection type.
+ *
+ * @param {Object} params - The parameters object
```

Comment:
suggestion (non-blocking): There's a lot of duplication with all of these param objects.

We could consider making a dedicated `interface` for these params, and moving all the comments to all of the fields of the params object. That will make the code more clear as the function signature looks like `{ config, integrations, ... }: IntegrationInstallParams`.